cpp/models/lct_secir/parameters_io.h

```diff
@@ -0,0 +1,317 @@
+                    }
+                }
+            }
+
+            // Compute initial values for compartment InfectedSevere.
+            if (offset >= std::floor(-timeInfectedSymptoms - timeInfectedSevere) &&
+                offset <= std::ceil(-timeInfectedSymptoms)) {
+                ScalarType THi =
+                    timeInfectedSevere /
+                    (ScalarType)LctState::template get_num_subcompartments<InfectionState::InfectedSevere>();
+                ScalarType mu_IH = model.parameters.template get<SeverePerInfectedSymptoms>();
+                int idx_H1       = LctState::template get_first_index<InfectionState::InfectedSevere>();
+                for (int i = 0; i < (int)LctState::template get_num_subcompartments<InfectionState::InfectedSevere>();
+                     i++) {
+                    if (offset == std::floor(-timeInfectedSymptoms - THi * (i + 1))) {
+                        init[idx_H1 + i] -= mu_IH *
+                                            (1 - (-timeInfectedSymptoms - (i + 1) * THi -
+                                                  std::floor(-timeInfectedSymptoms - (i + 1) * THi))) *
+                                            scale_confirmed_cases * entry.num_confirmed;
+                    }
+                    if (offset == std::ceil(-timeInfectedSymptoms - THi * (i + 1))) {
+                        init[idx_H1 + i] -= mu_IH *
+                                            (-timeInfectedSymptoms - (i + 1) * THi -
+                                             std::floor(-timeInfectedSymptoms - (i + 1) * THi)) *
+                                            scale_confirmed_cases * entry.num_confirmed;
+                    }
+                    if (offset == std::floor(-timeInfectedSymptoms - THi * i)) {
+                        init[idx_H1 + i] +=
+                            mu_IH *
+                            (1 - (-timeInfectedSymptoms - i * THi - std::floor(-timeInfectedSymptoms - i * THi))) *
+                            scale_confirmed_cases * entry.num_confirmed;
+                    }
+                    if (offset == std::ceil(-timeInfectedSymptoms - THi * i)) {
+                        init[idx_H1 + i] +=
+                            mu_IH * (-timeInfectedSymptoms - i * THi - std::floor(-timeInfectedSymptoms - i * THi)) *
+                            scale_confirmed_cases * entry.num_confirmed;
+                    }
+                }
+            }
+
+            // Compute initial values for compartment InfectedCritical.
+            if (offset >= min_offset_needed && offset <= std::ceil(-timeInfectedSymptoms - timeInfectedSevere)) {
+                ScalarType TUi = timeInfectedCritical /
+                                 LctState::template get_num_subcompartments<InfectionState::InfectedCritical>();
+                ScalarType mu_IH = model.parameters.template get<SeverePerInfectedSymptoms>();
```

Comment:
Here you are using the SECIHURD names for the compartments although otherwise in the code you are using InfectedNoSymptoms etc. This also holds before when using C and E etc. above. This might be confusing if one only knows the code and not your thesis.